lib/vector-common/src/sensitive_string.rs

```diff
@@ -0,0 +1,67 @@
+use vector_config::{configurable_component, ConfigurableString};
+
+/// Wrapper for sensitive strings containing credentials
+#[configurable_component(no_deser, no_ser)]
+#[cfg_attr(
+    feature = "serde",
+    derive(::serde::Deserialize, ::serde::Serialize),
+    serde(from = "String", into = "String")
+)]
+#[derive(Clone, Default, PartialEq, Eq)]
```

Comment:
You'll want to add something like this to the struct:

```rust
#[configurable(metadata(sensitive))]
pub struct SensitiveString(String);
```

This will be exposed in the configuration schema, allowing someone to programmatically detect which fields should be treated as sensitive strings.

Reply:
If I do so, I get a weird message 🤔 
```
failed to resolve: could not find `vector_config_common` in the list of imported crates
could not find `vector_config_common` in the list of imported crates
```